### PR TITLE
Allow interpolated values to be processed as YAML literals.

### PR DESCRIPTION
**Pull request by the-allanc**
This change does alter the behaviour of some of the original tests - for the better, in my opinion, but a change in behaviour nonetheless.

### CODE REVIEW

**Comment by lbolla** [comment]
Can you please also add a comment in README, saying that env vars will be treated as YAML entities?

File: yamlenv/env.py [hunk 1]
```diff
@@ -79,5 +80,5 @@ def interpolate(data):
             x = data
             for k in path[:-1]:
                 x = x[k]
-            x[path[-1]] = e.value
+            x[path[-1]] = yaml.load(e.value)
```
Comment (lbolla) [inline yamlenv/env.py]: Should this be `yaml.safe_load`, for added security?

Comment (lbolla) [reply yamlenv/env.py]: I wonder if it's cleaner to have `EnvVar` expose an attributed like `yaml_value` that just calls `yaml.safe_load`: `interpolate()` at the moment seems to be doing too many things.

Comment (the-allanc) [reply yamlenv/env.py]: > Should this be `yaml.safe_load`, for added security?

I don't personally mind, although you might then want to consider the use of `safe_load` in `yamlenv.Loader.construct_include` for consistency sake.

**Pull request merged**